Zend/tests/gh14009_004.phpt

```diff
@@ -0,0 +1,37 @@
+--TEST--
+GH-14009: Traits inherit prototype
+--FILE--
+<?php
+
+class A {
+    protected function test() {
+        throw new Exception('Unreachable');
+    }
+}
+
+class B extends A {
+    // Prototype is A::test().
+    protected function test() {
+        echo __METHOD__, "\n";
+    }
+}
+
+trait T {
+    protected abstract function test();
```

Comment:
What is the purpose? - I would expect an exception the same as if declared in the class directly.

Reply:
https://3v4l.org/N5otgm

Reply:
Traits support abstract methods. They are not equivalent to copy/pasting the method into the class.

https://www.php.net/manual/en/language.oop5.traits.php#language.oop5.traits.abstract